module/zfs/zio.c

```diff
@@ -3039,6 +3050,36 @@ zio_nop_write(zio_t *zio)
 	return (zio);
 }
 
+/*
+ * ==========================================================================
+ * Block Reference Table
+ * ==========================================================================
+ */
+static zio_t *
+zio_brt_free(zio_t *zio)
+{
+	blkptr_t *bp;
+
+	bp = zio->io_bp;
+
+	if (BP_GET_LEVEL(bp) > 0) {
+		return (zio);
+	}
+	if (BP_IS_METADATA(bp)) {
+		return (zio);
+	}
+
+	if (!brt_entry_decref(zio->io_spa, bp)) {
```

Comment:
It looks like we're able to reliably trigger a VERIFY here while running the `tests/functional/no_space` tests.

http://build.zfsonlinux.org/builders/CentOS%208%20x86_64%20%28TEST%29/builds/11642/steps/shell_4/logs/tests

```
[ 9291.037879] VERIFY(brtvd != NULL) failed
[ 9291.042240] PANIC at brt.c:1471:brt_entry_decref()
[ 9291.046671] Showing stack for process 1786796
[ 9291.050952] CPU: 0 PID: 1786796 Comm: z_fr_iss Tainted: P           OE    --------- -  - 4.18.0-425.13.1.el8_7.x86_64 #1
[ 9291.060262] Hardware name: Amazon EC2 m5d.large/, BIOS 1.0 10/16/2017
[ 9291.065472] Call Trace:
[ 9291.068742]  dump_stack+0x41/0x60
[ 9291.072413]  spl_panic+0xd0/0xe8 [spl]
[ 9291.093386]  spl_assert.constprop.3+0x16/0x20 [zfs]
[ 9291.098315]  brt_entry_decref+0x225/0x2b0 [zfs]
[ 9291.102987]  zio_brt_free+0x4a/0x80 [zfs]
[ 9291.107370]  zio_execute+0xee/0x210 [zfs]
[ 9291.111963]  taskq_thread+0x203/0x420 [spl]
[ 9291.130092]  kthread+0x10b/0x130
```

Reply:
I was able to reproduce it. Working on a fix. Thanks.

Reply:
Fixed.